### PR TITLE
zld: 1.3.3 -> 1.3.4

### PR DESCRIPTION
**Pull request by astutecat**
###### Description of changes

Update to zld from 1.3.3 to 1.3.4. This version includes a fix to an issue where, during Rust builds, executable produced would crash with code signing issue (work around of Apple codesigning issue)

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/tools/zld/default.nix [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 stdenv.mkDerivation rec {
   pname = "zld";
-  version = "1.3.3";
+  version = "1.3.4";
   src = fetchzip {
     url = "https://github.com/michaeleisel/zld/releases/download/${version}/zld.zip";
     sha256 = "0qb4l7a4vhpnzkgzhw0jivz40jr5gdhqfyynhbkhn7ryh5s52d1p";
```
Comment (SuperSandro2000) [inline pkgs/development/tools/zld/default.nix:8]: You also must update the hash.

Comment (astutecat) [reply pkgs/development/tools/zld/default.nix]: whoops, sorry about that - I've updated and pushed.

**Comment by superherointj** [comment]
Squash commits.

**Comment by superherointj** [comment]
Result of `nixpkgs-review pr 188830` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)

**Comment by astutecat** [comment]
> Please, squash commits.

done :)

**Comment by superherointj** [comment]
Result of `nixpkgs-review pr 188830` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)

**Review by superherointj — APPROVED** [review]

**Pull request merged**